Sources/FoundationInternationalization/Locale/Locale_ICU.swift

```diff
@@ -1229,6 +1230,115 @@ internal final class _LocaleICU: _LocaleProtocol, Sendable {
         }
     }
 
+    var weekendRange: WeekendRange? {
+        let firstWeekday = self.firstDayOfWeek
+        return lock.withLock { state -> WeekendRange? in
+
+            if let r = state.weekendRange {
+                return r
+            }
+
+            var result = WeekendRange(start: 0, end: 0)
+
+            var weekdaysIndex : [UInt32] = [0, 0, 0, 0, 0, 0, 0]
+            weekdaysIndex[0] = UInt32(firstWeekday.icuIndex)
+            for i in 1..<7 {
+                weekdaysIndex[i] = (weekdaysIndex[i - 1] % 7) + 1
+            }
+
+            var weekdayTypes : [UCalendarWeekdayType] = [UCAL_WEEKDAY, UCAL_WEEKDAY, UCAL_WEEKDAY, UCAL_WEEKDAY, UCAL_WEEKDAY, UCAL_WEEKDAY, UCAL_WEEKDAY]
+
+            var onset: UInt32?
+            var cease: UInt32?
+
+            var status = U_ZERO_ERROR
+            let cal = ucal_open(nil, 0, identifier, UCAL_DEFAULT, &status)
```

Comment:
Is it going to be more expensive (CPU time) to open and close the calendar here, vs on `Calendar_ICU` where we reuse the `ucalendar` we already had?

I don't think it is a showstopper, because this makes a lot more sense, but we should be aware.

Reply:
I don't see an obvious performance regression. This value is cached inside this `_LocaleICU` instance. So we'd only call `ucal_open` once per Calendar-Locale combination. Since we're also caching locale by its identifier, we'd be fetching from the cache most of the time.